### PR TITLE
COMP-707 Update tox configuration

### PR DESCRIPTION
**Pull request by rbrazinskas**
- Supported Python interpreter versions are "pinned down" to 3.9 and 3.10
- Tox skips missing versions of Python interpreters (instead of raising an exception)
- Project is migrated to use <pyproject.toml> file as a single and only source of configuration (except of <AUTHORS.rst> file which is needed for document build)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,23 +1,182 @@
 [build-system]
-requires = ["setuptools>=46.1.0", "setuptools_scm[toml]>=5", "wheel"]
+requires = ["setuptools>=65.4.1", "setuptools_scm[toml]>=7", "wheel"]
```
Comment (smite) [inline pyproject.toml:2]:
```suggestion
requires = ["setuptools>=65.4.1", "setuptools_scm[toml]>=7"]
```
I think wheel isn't required here.

File: pyproject.toml [hunk 2]
```diff
@@ -1,23 +1,182 @@
...
 build-backend = "setuptools.build_meta"
 
+
+[project]
+name = "iqm-cortex_cli"
```
Comment (smite) [inline pyproject.toml:7]:
```suggestion
name = "iqm-cortex-cli"
```
For consistency, package names are normalized: https://packaging.python.org/en/latest/specifications/name-normalization/

File: pyproject.toml [hunk 3]
```diff
@@ -1,23 +1,182 @@
...
+[tool.setuptools]
+platforms = ["any"]
+zip-safe = false
+include-package-data = true
+package-dir = {"" = "src"}
+
+[tool.setuptools.packages.find]
+where = ["src"]
+exclude = ["tests"]
+
```
Comment (smite) [inline pyproject.toml:79]:
```suggestion
```
I think all of these are unnecessary, the default setuptools file discovery method should work fine with our project setup.

File: setup.py [hunk 4]
```diff
@@ -1,21 +1,4 @@
...
-        setup(use_scm_version={"version_scheme": "no-guess-dev"})
-    except:  # noqa
-        print(
-            "\n\nAn error occurred while building the project, "
-            "please ensure you have the most updated version of setuptools, "
-            "setuptools_scm and wheel with:\n"
-            "   pip install -U setuptools setuptools_scm wheel\n\n"
-        )
-        raise
+    setup()
```
Comment (smite) [inline setup.py:4]: Can't we just remove `setup.py` entirely?

File: pyproject.toml [hunk 5]
```diff
@@ -1,23 +1,182 @@
...
+dynamic = ["version"]
+description = "CLI for executing quantum circuits"
+authors = [{name = "IQM Finland Oy", email = "developers@meetiqm.com"}]
+readme = "README.rst"
+license = {file = "LICENSE.rst"}
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Programming Language :: Python :: 3 :: Only",
+    "Topic :: Scientific/Engineering :: Physics",
+    "Intended Audience :: Science/Research",
```
[the remark this answers is not in the document]
Comment (rbrazinskas) [reply pyproject.toml:17]: Any suggestions what classifiers to use?

File: setup.py (hunk 4 above)
Comment (rbrazinskas) [reply setup.py:4]: I was not sure... I checked EXA repositories and consulted Ozzy, they have dummy ```setup.py``` everywhere:

"Unfortunately, the lowest pip and python versions we support require a dummy setup.py for pip install to work. We'll re-evaluate after finishing the 3.11 migration, but we'll probably have to keep them around for another minor version or two, until we mostly drop 3.9 support"

Any idea how can I test this before merging?

Comment (rbrazinskas) [reply setup.py]: I have removed ```setup.py``` after testing build and installation with latest ```pip``` versions

File: pyproject.toml (hunk 5 above)
[the remark this answers is not in the document]
Comment (smite) [reply pyproject.toml:17]: I guess they're fine for now.

File: pyproject.toml [hunk 6]
```diff
@@ -1,23 +1,172 @@
...
+homepage = "https://github.com/iqm-finland/cortex-cli"
+documentation = "https://iqm-finland.github.io/cortex-cli"
+repository = "https://github.com/iqm-finland/cortex-cli.git"
+changelog = "https://github.com/iqm-finland/cortex-cli/blob/main/CHANGELOG.rst"
+
+[project.optional-dependencies]
+# Add here additional requirements for extra features, to install with:
+# `pip install iqm-cortex-cli[testing,docs,dev]`
+circuit = [
+    "cirq-iqm >= 11.0, < 12.0",
```
Comment (smite) [inline pyproject.toml:40]:
```suggestion
    "cirq-iqm >= 11.7, < 12.0",
```

File: pyproject.toml [hunk 7]
```diff
@@ -1,23 +1,172 @@
...
+
+[project.scripts]
+cortex = "cortex_cli.cortex_cli:cortex_cli"
+
+
+
 [tool.setuptools_scm]
-# See configuration details in https://github.com/pypa/setuptools_scm
 version_scheme = "no-guess-dev"
+write_to = "version.txt"
```
Comment (smite) [inline pyproject.toml:73]: Is this necessary? Do the CI scripts use this file?

File: pyproject.toml [hunk 8]
```diff
@@ -1,23 +1,172 @@
...
+]
+
 
 [tool.black]
 line-length = 120
 skip-string-normalization = true
-target-version = ['py311']
-extend-exclude = '_pb2.py$'
+target-version = ["py311"]
+extend-exclude = "_pb2.py$"
```
Comment (smite) [inline pyproject.toml:97]:
```suggestion
```
We have no protobuf files here, right?

**Review by smite — APPROVED** [review]

**Review by smite — DISMISSED** [review]

**Review by freetonik — APPROVED** [review]

**Pull request merged**